textinput/textinput.go

```diff
@@ -91,6 +92,66 @@ func (c CursorMode) String() string {
 	}[c]
 }
 
+// ValidateBool validates boolean-like inputs only
+// i.e 'y', 'n', 't', 'f', '1', '0' (case insensitive)
+func ValidateBool(runes []rune, m Model) []rune {
+	out := make([]rune, 0)
+
+	for _, r := range runes {
+		if len(m.value) <= 0 {
+			_, parseErr := strconv.ParseBool(string(r))
+			if parseErr == nil || r == 'y' || r == 'Y' || r == 'n' || r == 'N' {
+				out = append(out, r)
+			}
+		}
+	}
+
+	return out
+}
+
+// ValidateInt validates integer-only inputs.
+func ValidateInt(runes []rune, m Model) []rune {
+	out := make([]rune, 0)
+	for _, r := range runes {
+		_, parseErr := strconv.ParseInt(string(r), 10, 32)
+		if parseErr == nil {
+			out = append(out, r)
+		}
+	}
+
+	return out
+}
+
+// ValidateFloat validates float-only inputs.
+func ValidateFloat(runes []rune, m Model) []rune {
+	out := make([]rune, 0)
```

Comment:
Any issue with using https://pkg.go.dev/strconv#ParseFloat here as well?

Reply:
I'm not sure if there's any advantage to using that as opposed to ParseInt since we're parsing a single digit every time

Reply:
I see, I read that change wrong. I would pass the entire content of the `textinput` to the Validate method whenever its content changes. There are too many different ways of manipulating its content, I fear the code is becoming too convoluted quickly: entering characters, entering characters after moving the cursor, the clipboard, changing it with `textinput`'s API.

Reply:
Right. I'm not sure if refactoring that is within the scope of this PR, but I could give it a try and open a separate PR before having this merged